lib/index.js

```diff
@@ -49,15 +53,34 @@ module.exports = {
 		const folder = path.dirname(currentFileName);
 		const file = path.basename(currentFileName);
 
-		this.panel.renderStartProcess(file);
+		this.panel.renderStartProcess();
 		this.testRunnerProcess.run(folder, file);
 	},
-	toggle() {
+	runAll() {
+		if (!this.atomPanel.isVisible()) {
+			this.toggle(this._runAll);
+		}
+
+		if (!this.canRun()) {
+			return;
+		}
+
+		const editor = atom.workspace.getActiveTextEditor();
+		const currentFileName = editor.buffer.file.path;
+		const paths = editor.project.rootDirectories.map(d => d.path);
+		const selectedProject = paths.filter(p => currentFileName.startsWith(p));
```

Comment:
Not sure that's equivalent.

As you said, the first line is terribly complicated and `editor.project.getPaths();` should do it. But the second line is there to infer the selected project from the currently selected file, as Atom supports different projects.

Does that make sense?. getPaths() does not give you the selected or active project.